pkg/container-disk/container-disk_test.go

```diff
@@ -181,7 +181,29 @@ var _ = Describe("ContainerDisk", func() {
 					},
 				}
 				containers := GenerateContainers(vmi, "libvirt-runtime", "/var/run/libvirt")
-				Expect(containers[0].Resources.Limits).To(HaveLen(2))
+
+				containerResourceSpecs := []k8sv1.ResourceList{containers[0].Resources.Limits, containers[0].Resources.Requests}
+
+				for _, containerResourceSpec := range containerResourceSpecs {
+					Expect(containerResourceSpec).To(And(HaveKey(k8sv1.ResourceCPU), HaveKey(k8sv1.ResourceMemory)))
```

Comment:
This is a test I did not write. The test verifies that the resources that were explicitly defined in VMI (in this case memory and CPU request/limit) are indeed defined in container specs. The original test implementation expected the length resource limits/requests to be 2 (memory & cpu), therefore the test failed since ephemeral memory is now also requested, changing the length to be 3 (memory, cpu and ephemeral storage).

It wasn't my intention to change the goal of the test which has nothing to do with ephemeral storage but with the explicitly "user-defined" resources. So this change only caused the test to more accurate.

Another test dedicated for ephemeral storage is also added.

Reply:
I don't mind this test being as it is now. As long as we have another test that explicitly checks the ephemeral storage, and it seems that we do. so it looks good to me.